### PR TITLE
Delete all associated data together with user (GSI-1050)

### PR DESCRIPTION
**Pull request by Cito**
The user registry has end endpoint to delete users. So far this removed only the user itself and any IVAs belonging to the user. The TOTP tokens (maintained by the auth adapter) and the claims (maintained by the claims repository) remained as orphaned datasets. I have added a fix for this here, together with an integration test.

### CODE REVIEW

File: tests/integration/user_management/user_registry/test_api.py [hunk 1]
```diff
@@ -783,6 +786,85 @@ async def test_delete_user_unauthenticated(bare_client: BareClient):
...
+    for collection in collections:
+        assert num_documents(collection) == len(user_ids), collection
+
+    # successively delete the users and check that all associated data is deleted
+    while user_ids:
+        id_ = user_ids.pop()
+        response = await full_client.delete(f"/users/{id_}", headers=steward_headers)
+        assert response.status_code == status.HTTP_204_NO_CONTENT
+        bad_collections = set(collections)
+        # try a couple of times because to allow for processing
```
Comment (Cito) [inline tests/integration/user_management/user_registry/test_api.py:857]:
```suggestion
        # try a couple of times to allow for processing time
```

File: tests/integration/user_management/user_registry/test_api.py [hunk 2]
```diff
@@ -783,6 +786,85 @@ async def test_delete_user_unauthenticated(bare_client: BareClient):
...
+
+    # successively delete the users and check that all associated data is deleted
+    while user_ids:
+        id_ = user_ids.pop()
+        response = await full_client.delete(f"/users/{id_}", headers=steward_headers)
+        assert response.status_code == status.HTTP_204_NO_CONTENT
+        bad_collections = set(collections)
+        # try a couple of times to allow for processing time
+        for _tries in range(100):
+            for collection in list(bad_collections):
```
Comment (TheByronHimes) [inline tests/integration/user_management/user_registry/test_api.py:859]: Why not iterate through the tuple already declared above?
```suggestion
            for collection in collections:
```

**Review by TheByronHimes — APPROVED** [review]
Looks good to me

File: tests/integration/user_management/user_registry/test_api.py (hunk 2 above)
Comment (Cito) [reply tests/integration/user_management/user_registry/test_api.py:859]: Since we only need to test the collections that have not yet been tested.

**Pull request merged**